earth_enterprise/src/server/geresetpgdb

```diff
@@ -201,27 +207,49 @@ if [ -e $VARPGDIR/data/postmaster.pid ]; then
 fi
 
 if [ $type == "backup" ]; then
-  if [ ! -e $VARPGDIR/data ]; then
-    echo "WARNING: There is no data to back up."
-  else
-    do_dump
-  fi
-  exit
+  do_dump
 fi
 
 # if the data dir does not exist then this has to be a hard reset.
+# TODO: This will need to be removed when we unbundle
 if [ ! -e $VARPGDIR/data ]; then
   type="hard";
 fi
 
 if [ $type == "hard" ]; then
-  # Do reset and recreate database instance.
-  do_reset
+  if [ ! -e $VARPGDIR/data ]; then
+    # create a new db instance from scratch
+    # TODO: This will need to be removed when we unbundle
+    $BINDIR/initdb --auth=trust -D $VARPGDIR/data >/dev/null
+  else
+    # Start postmaster.
+    if [ ! -f $VARPGDIR/data/postmaster.pid ]; then
+      $BINDIR/pg_ctl -D $VARPGDIR/data -l $VARPGDIR/logs/pg.log start -w
+    fi
+
+    # Do reset and recreate database instance.
+    echo Removing geserve-databases...
+    do_reset
+
+    # Also drop example databases
```

Comment:
Any reason why 'geplaces; and 'searchexample' dropping was not made part of 'do_reset' ?

Reply:
To match the same behavior as before, that is done as part of the hard reset, not the soft reset. My guess is performance is the reason behind it. Soft reset is used with the upgrade target and it takes up to a minute to reinstall the example databases.